env0/resource_environment.go

```diff
@@ -156,6 +296,85 @@ func getUpdatePayload(d *schema.ResourceData) client.EnvironmentUpdate {
 	return payload
 }
 
+func getDeployPayload(d *schema.ResourceData) client.DeployRequest {
+	payload := client.DeployRequest{}
+
+	if templateId, ok := d.GetOk("template_id"); ok {
+		payload.BlueprintId = templateId.(string)
+	}
+
+	if revision, ok := d.GetOk("revision"); ok {
+		payload.BlueprintRevision = revision.(string)
+	}
+
+	if configuration, ok := d.GetOk("configuration"); ok {
+		configurationChanges := getConfigurationVariables(configuration.([]interface{}))
+		payload.ConfigurationChanges = &configurationChanges
+	}
+
+	if ttl, ok := d.GetOk("ttl"); ok {
+		payload.TTL = &client.TTL{
+			Type:  ttl.(map[string]interface{})["type"].(string),
+			Value: ttl.(map[string]interface{})["value"].(string),
```

Comment:
Would be cool if you can define a struct for the TTL schema...

Reply:
deleted it here https://github.com/env0/terraform-provider-env0/pull/174 😄